src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/Auxiliary2AzureSearchCommand.cs

```diff
@@ -1,17 +1,287 @@
+        private readonly IOptionsSnapshot<Auxiliary2AzureSearchConfiguration> _options;
+        private readonly IAzureSearchTelemetryService _telemetryService;
+        private readonly ILogger<Auxiliary2AzureSearchCommand> _logger;
+
+        public Auxiliary2AzureSearchCommand(
+            IAuxiliaryFileClient auxiliaryFileClient,
+            IDownloadDataClient downloadDataClient,
+            IDownloadSetComparer downloadSetComparer,
+            ISearchDocumentBuilder searchDocumentBuilder,
+            ISearchIndexActionBuilder indexActionBuilder,
+            Func<IBatchPusher> batchPusherFactory,
+            ISystemTime systemTime,
+            IOptionsSnapshot<Auxiliary2AzureSearchConfiguration> options,
+            IAzureSearchTelemetryService telemetryService,
+            ILogger<Auxiliary2AzureSearchCommand> logger)
+        {
+            _auxiliaryFileClient = auxiliaryFileClient ?? throw new ArgumentNullException(nameof(auxiliaryFileClient));
+            _downloadDataClient = downloadDataClient ?? throw new ArgumentNullException(nameof(downloadDataClient));
+            _downloadSetComparer = downloadSetComparer ?? throw new ArgumentNullException(nameof(downloadSetComparer));
+            _searchDocumentBuilder = searchDocumentBuilder ?? throw new ArgumentNullException(nameof(searchDocumentBuilder));
+            _indexActionBuilder = indexActionBuilder ?? throw new ArgumentNullException(nameof(indexActionBuilder));
+            _batchPusherFactory = batchPusherFactory ?? throw new ArgumentNullException(nameof(batchPusherFactory));
+            _systemTime = systemTime ?? throw new ArgumentNullException(nameof(systemTime));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            if (_options.Value.MaxConcurrentBatches <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(options),
+                    $"The {nameof(AzureSearchJobConfiguration.MaxConcurrentBatches)} must be greater than zero.");
+            }
+
+            if (_options.Value.MaxConcurrentVersionListWriters <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(options),
+                    $"The {nameof(AzureSearchJobConfiguration.MaxConcurrentVersionListWriters)} must be greater than zero.");
+            }
+        }
+
         public async Task ExecuteAsync()
         {
-            // TODO: implement the job
-            // https://github.com/NuGet/NuGetGallery/issues/6447
-            await Task.Yield();
+            var stopwatch = Stopwatch.StartNew();
+            var success = false;
+            try
+            {
+                _logger.LogInformation("Fetching old download count data from blob storage.");
```

Comment:
Do you think it's worth adding a comment that `newData` comes from statistics pipeline and `oldResult` is the last indexed download counts? I was a little confused when I first saw this

Reply:
Yes, that's a good idea.